cmd/kubeadm/app/cmd/init.go

```diff
@@ -145,14 +142,9 @@ func NewCmdInit(out io.Writer) *cobra.Command {
 		"skip preflight checks normally run before modifying the system",
 	)
 
-	cmd.PersistentFlags().Int32Var(
-		&cfg.API.BindPort, "api-port", cfg.API.BindPort,
-		"Port for API to bind to",
-	)
-
-	cmd.PersistentFlags().Int32Var(
-		&cfg.Discovery.BindPort, "discovery-port", cfg.Discovery.BindPort,
-		"Port for JWS discovery service to bind to",
+	cmd.PersistentFlags().Var(
+		discovery.NewDiscoveryValue(&cfg.Discovery), "discovery",
+		"Discovery",
```

Comment:
`/Discovery/The discovery method kubeadm will use for connecting nodes to the master/` or similar